File: Chapter 10/SubwayLoader.java
package chapter10;

import java.io.File;
import java.io.*;

public class SubwayLoader
{
    private Subway subway;
    
    public SubwayLoader() {
        this.subway = new Subway();
    }
    
    public Subway loadFromFile(File subwayFile) throws IOException {
        BufferedReader reader = new BufferedReader(new FileReader(subwayFile));
        loadStations(subway, reader);
        String lineName = reader.readLine();
        while ((lineName != null) && (lineName.length() > 0)) {
            loadLine(subway, reader, lineName);
            lineName = reader.readLine();
        }
        return subway;
    }
    
    private void loadStations(Subway subway, BufferedReader reader) throws IOException {
        String currentLine;
        currentLine = reader.readLine();
        while (currentLine.length() > 0) {
            subway.addStation(currentLine);
            currentLine = reader.readLine();
        }
    }
    
    private void loadLine(Subway subway, BufferedReader reader, String lineName) throws IOException {
        String station1Name, station2Name;
        station1Name = reader.readLine();
        station2Name = reader.readLine();
        while ((station2Name != null) && (station2Name.length() > 0)) {
            subway.addConnection(station1Name, station2Name, lineName);
            station1Name = station2Name;
            station2Name = reader.readLine();
        }
    }
}

File: Ch 10 - The OOA&D Lifecycle/Iteration 2 - Objectville Subway/Loader/SubwayLoader.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using Objectville_Subway.SubwaySystem;

namespace Objectville_Subway.Loader
{
    class SubwayLoader
    {
        private Subway subway = new Subway();

        public SubwayLoader()
        {

        }

        public Subway loadFromFile(string subwayFilePath)
        {
            using (StreamReader reader = new StreamReader(subwayFilePath))
            {
                loadStations(subway, reader);
                string lineName = reader.ReadLine();
                while ((lineName != null) && (lineName.Length > 0))
                {
                    loadLine(subway, reader, lineName);
                    lineName = reader.ReadLine();
                }
            }
            return subway;
        }

        private void loadStations(Subway subway, StreamReader reader)
        {
            string currentLine = reader.ReadLine();
            while (currentLine.Length > 0)
            {
                subway.AddStation(currentLine);
                currentLine = reader.ReadLine();
            }
        }

        private void loadLine(Subway subway, StreamReader reader, string lineName)
        {
            string station1Name = reader.ReadLine();
            string station2Name = reader.ReadLine();
            while ((station2Name != null) && (station2Name.Length > 0))
            {
                subway.AddConnection(lineName, station1Name, station2Name);
                station1Name = station2Name;
                station2Name = reader.ReadLine();
            }

        }
    }
}
